akka-docs/src/main/paradox/project/immutable.md

```diff
@@ -0,0 +1,87 @@
+---
+project.description: Data immutability using Project Lombok
+---
+# Immutability using Lombok
+
+A preferred best practise in akka is to have immutable messages. Scala provides case class which makes it extremely easy
+to have short and clean classes for creating immutable objects, but no such facility is easily available in Java. We can make use
+of several third party libraries which help is achieving this. One good example is Lombok.
+
+Project Lombok is a java library that automatically plugs into your editor and build tools, and helps 
+get rid of much of the boilerplate code for java development.
+
+Lombok handles the following details for you. It:
+
+* modifies fields to be `private` and `final`
+* creates getters for each field
+* creates correct `equals`, `hashCode` and a human-friendly `toString`
+* creates a constructor requiring all fields.
+
+
+### Adding Lombok to your project
+To add Lombok to a Maven project, declare it as a simple dependency:
+
+Maven
+: @@dependency {
+    <dependency>
+        <groupId>org.projectlombok</groupId>
+        <artifactId>lombok</artifactId>
+        <version>1.18.10</version>
+    </dependency>
+}
+
+Gradle
+: @@dependency {
+    dependencies {
+    	compileOnly 'org.projectlombok:lombok:1.18.10'
+    	annotationProcessor 'org.projectlombok:lombok:1.18.10'
+    }
+} 
+    
+    
+# Using lombok    
+
+    @Value
+    public class LombokUser {
+    
+      String name;
+    
+      String email;
+    }
```

Comment:
we normally include code from compiled sources, but let's make an exception here to avoid adding complexity to the build